docs/apis/http/oidc/_index.md

```diff
@@ -0,0 +1,109 @@
+---
+title: "OIDC"
+date: 2023-10-10T00:00:00+00:00
+weight: 21
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/apis/http/idp
+geekdocFilePath: _index.md
+geekdocCollapseSection: true
+---
+
+Infinite Scale can act as an identity provider (IdP) for OpenID Connect (OIDC) authentication .
+OIDC defines a discovery mechanism, called OpenID Connect Discovery,
+where an OpenID server publishes its metadata at a well-known URL, typically:
+
+`https://ocis.test/.well-known/openid-configuration`
+
+This URL returns a JSON listing of the OpenID/OAuth endpoints, supported scopes and claims, public keys used to sign the tokens, and other details.
+The clients can use this information to construct a request to the OpenID server.
+The field names and values are defined in the [OpenID Connect Discovery Specification](https://openid.net/specs/openid-connect-discovery-1_0.html).
+Here is an example of data returned:
+```json
+{
+  "issuer": "https://ocis.test:9200",
+  "authorization_endpoint": "https://ocis.test:9200/signin/v1/identifier/_/authorize",
+  "token_endpoint": "https://ocis.test:9200/konnect/v1/token",
+  "userinfo_endpoint": "https://ocis.test:9200/konnect/v1/userinfo",
+  "end_session_endpoint": "https://ocis.test:9200/signin/v1/identifier/_/endsession",
+  "check_session_iframe": "https://ocis.test:9200/konnect/v1/session/check-session.html",
+  "jwks_uri": "https://ocis.test:9200/konnect/v1/jwks.json",
+  "scopes_supported": [
+    "openid",
+    "offline_access",
+    "profile",
+    "email",
+    "LibgreGraph.UUID",
+    "LibreGraph.RawSub"
+  ],
+  "response_types_supported": [
+    "id_token token",
+    "id_token",
+    "code id_token",
+    "code id_token token"
+  ],
+  "subject_types_supported": [
+    "public"
+  ],
+  "id_token_signing_alg_values_supported": [
+    "RS256",
+    "RS384",
+    "RS512",
+    "PS256",
+    "PS384",
+    "PS512"
+  ],
+  "userinfo_signing_alg_values_supported": [
+    "RS256",
+    "RS384",
+    "RS512",
+    "PS256",
+    "PS384",
+    "PS512"
+  ],
+  "request_object_signing_alg_values_supported": [
+    "ES256",
+    "ES384",
+    "ES512",
+    "RS256",
+    "RS384",
+    "RS512",
+    "PS256",
+    "PS384",
+    "PS512",
+    "none",
+    "EdDSA"
+  ],
+  "token_endpoint_auth_methods_supported": [
+    "client_secret_basic",
+    "none"
+  ],
+  "token_endpoint_auth_signing_alg_values_supported": [
+    "RS256",
+    "RS384",
+    "RS512",
+    "PS256",
+    "PS384",
+    "PS512"
+  ],
+  "claims_parameter_supported": true,
+  "claims_supported": [
+    "iss",
+    "sub",
+    "aud",
+    "exp",
+    "iat",
+    "name",
+    "family_name",
+    "given_name",
+    "email",
+    "email_verified"
+  ],
+  "request_parameter_supported": true,
+  "request_uri_parameter_supported": false
+}
+```
+
+In OIDC, authentication can follow one of the three methods described on the [official site](https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.3):
```

Comment:
Should the auth flow be specific to ocis? (because oidc can have many more)
```suggestion
In Infinite Scale, authentication can follow one of the three methods described on the [official site](https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.3):
```